lints/lint_subject_common_name_not_from_san.go

```diff
@@ -40,7 +42,7 @@ func (l *subjectCommonNameNotFromSAN) Execute(c *x509.Certificate) *LintResult {
 	cn := c.Subject.CommonName
 
 	for _, dn := range c.DNSNames {
-		if cn == dn {
+		if strings.ToLower(cn) == strings.ToLower(dn) {
```

Comment:
Does `ToLower` comparison work for all legal Fully-Qualfied Domain Names? It sounds like [EqualFold](https://golang.org/pkg/strings/#EqualFold) may be the appropriate comparator here -- 

> EqualFold reports whether s and t, interpreted as UTF-8 strings, are equal under Unicode case-folding.

```suggestion
		if strings.EqualFold(cn, dn) {
```

Reply:
tl;dr: there should only be a limited set of ASCII characters in these fields on CA/B compliant certs, therefore `strings.ToLower` should be fine.

Legal values in the `subjectAlternativeName` field are defined in the following CA/B Forum rule: https://github.com/cabforum/documents/blob/master/docs/BR.md#71421-subject-alternative-name-extension which reads:
```Each entry MUST be either a dNSName containing the Fully-Qualified Domain Name or an iPAddress containing the IP address of a server. ```
In RFC5280, `dNSName` is defined as: (https://tools.ietf.org/html/rfc5280#section-4.2.1.6)
```dNSName                         [2]     IA5String,```

And IA5String is defined as: https://en.wikipedia.org/wiki/IA5STRING
```generally equivalent to the first 128 characters of the ASCII alphabet```

According to https://tools.ietf.org/html/rfc5280#section-7.2, any international names must be converted to the `ASCII Compatible Encoding` format which is basically `xn--<punycode>` (https://en.wikipedia.org/wiki/Punycode).

Therefore, it stands to reason that the CN and subjectAlternativeNames fields should only have a limited set of ASCII characters available to them which should work with `strings.ToLower`.

Reply:
Looking at the two approaches in terms of performance, it seems like it makes sense to switch to the other function anyway. Thanks, @justinbastress